src/main/java/de/zalando/tip/zalenium/proxy/DockerSeleniumStarterRemoteProxy.java

```diff
@@ -515,7 +515,16 @@ private int findFreePortInRange(int lowerBoundary, int upperBoundary) throws Ill
                 }
             }
         }
-        throw new IllegalStateException("Could not find a free TCP/IP port to use for docker-selenium");
+        return -1;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) This class overrides "equals()" and should therefore also override "hashCode()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1206)